python/mxnet/model.py

```diff
@@ -69,15 +69,15 @@ def _create_sparse_kvstore(kvstore):
     update_on_kvstore : bool. Always True.
     """
     # always update on kvstore
-    update_on_kvstore = True
     if isinstance(kvstore, kvs.KVStore):
         kv = kvstore
     elif isinstance(kvstore, str):
         kv = kvs.create(kvstore)
     else:
         raise TypeError("Cannot create '%s' KVStore with row_sparse parameters. "
                         "The type must be KVStore or str." % kvstore)
-    return (kv, update_on_kvstore)
+    assert kv.is_capable(kvs.KVStoreBase.OPTIMIZER)
```

Comment:
Could you also add an error message: 
"KVStore with sparse weight requires optimizer support. However, type(kv) does not support optimizer. Please consider other kvstore backends (e.g. dist_device) instead."